v19.2/vectorized-execution.md

```diff
@@ -0,0 +1,58 @@
+
+With `vectorize=auto`, the `vectorize_row_count_threshold` [cluster setting](cluster-settings.html) specifies the minimum number of rows required to use the vectorized engine to execute a query plan. If the number of rows being processed falls below the row count threshold, CockroachDB uses the row-oriented execution engine instead. By default, `vectorize_row_count_threshold=1000`.
+
+## How vectorized execution works
+
+When you issue a SQL query in CockroachDB, the gateway node (i.e. the node from which you issue the query) [parses the query and creates a physical plan](architecture/sql-layer.html#sql-parser-planner-executor) for execution on each node. If vectorized execution is enabled, the physical plan is sent to each node, to be processed by the vectorized execution engine.
+
+Upon receiving [the physical plan](explain-analyze.html#distsql-plan-viewer), the vectorized engine reads batches of table data [from disk](architecture/storage-layer.html) and converts the data from row format to columnar format. These batches of column data are stored in-memory so that the engine can access them quickly during execution.
+
+To limit the complexity of query execution, the engine uses specialized, precompiled functions that quickly iterate over the type-specific arrays of column data. The columnar output from the functions is stored in memory as the engine processes each column of data. After processing all columns of data in the input buffer, the engine converts the columnar output back to row format, and then returns the processed rows to the SQL interface. After a batch of table data has been fully processed, the engine reads the following batch of table data for processing, until the query has been executed.
+
+To see a detailed view of the vectorized execution plan for a query, you can use the [`EXPLAIN(VEC)`](explain.html#vec-option) statement. For detailed examples of vectorized query execution for hash and merge joins, see the blog posts [40x faster hash joiner with vectorized execution](https://www.cockroachlabs.com/blog/vectorized-hash-joiner/) and [Vectorizing the merge joiner in CockroachDB](https://www.cockroachlabs.com/blog/vectorizing-the-merge-joiner-in-cockroachdb/).
+
+## Known Limitations
+
+The vectorized execution engine has the following limitations:
+
+- It is not as extensively tested as CockroachDB's existing row-oriented execution engine.
+- There is a risk that queries that perform large joins or other memory-intensive operations could cause the node to run out of memory and crash. As such, vectorized execution is experimental for non-streaming operations, including global sorts, hash joins, merge joins on non-unique columns, and window functions. We do not recommend using vectorized execution for non-streaming operations in production environments, as these operations can buffer an unlimited number of rows before they can start producing output.
```

Comment:
As mentioned previously, we should capture in one place (perhaps a new "Supported query types" section) what "streaming" and "non-streaming" mean, and all of the SQL they represent, with links to all such statements.  Perhaps then those could be linked from here via the term "streaming".